src/modules/utils/timing/Timestamp_fbt.cpp

```diff
@@ -0,0 +1,103 @@
+/*************************************************************************
+ *** FORTE Library Element
+ ***
+ *** This file was generated using the 4DIAC FORTE Export Filter V1.0.x NG!
+ ***
+ *** Name: Timestamp
+ *** Description: creates a Unix-Epoch-Timestamp in nanoseconds
+ *** Version:
+ ***     1.0: 2024-09-14/Monika Wenger -  -
+ *************************************************************************/
+
+#include "Timestamp_fbt.h"
+#ifdef FORTE_ENABLE_GENERATED_SOURCE_CPP
+#include "influxDB/Timestamp_fbt_gen.cpp"
+#endif
+
+#include "iec61131_functions.h"
+#include "forte_array_common.h"
+#include "forte_array.h"
+#include "forte_array_fixed.h"
+#include "forte_array_variable.h"
+
+DEFINE_FIRMWARE_FB(FORTE_influxDB__Timestamp, g_nStringIdinfluxDB__Timestamp)
+
+const CStringDictionary::TStringId FORTE_influxDB__Timestamp::scmDataOutputNames[] = {g_nStringIdtimestamp};
+const CStringDictionary::TStringId FORTE_influxDB__Timestamp::scmDataOutputTypeIds[] = {g_nStringIdULINT};
+const TForteInt16 FORTE_influxDB__Timestamp::scmEIWithIndexes[] = {-1};
+const CStringDictionary::TStringId FORTE_influxDB__Timestamp::scmEventInputNames[] = {g_nStringIdREQ};
+const TDataIOID FORTE_influxDB__Timestamp::scmEOWith[] = {0, scmWithListDelimiter};
+const TForteInt16 FORTE_influxDB__Timestamp::scmEOWithIndexes[] = {0};
+const CStringDictionary::TStringId FORTE_influxDB__Timestamp::scmEventOutputNames[] = {g_nStringIdCNF};
+const SFBInterfaceSpec FORTE_influxDB__Timestamp::scmFBInterfaceSpec = {
+  1, scmEventInputNames, nullptr, scmEIWithIndexes,
+  1, scmEventOutputNames, scmEOWith, scmEOWithIndexes,
+  0, nullptr, nullptr,
+  1, scmDataOutputNames, scmDataOutputTypeIds,
+  0, nullptr,
+  0, nullptr
+};
+
+FORTE_influxDB__Timestamp::FORTE_influxDB__Timestamp(const CStringDictionary::TStringId paInstanceNameId, forte::core::CFBContainer &paContainer) :
+    CFunctionBlock(paContainer, scmFBInterfaceSpec, paInstanceNameId),
+    var_conn_timestamp(var_timestamp),
+    conn_CNF(this, 0),
+    conn_timestamp(this, 0, &var_conn_timestamp) {
+};
+
+void FORTE_influxDB__Timestamp::setInitialValues() {
+  var_timestamp = 0_ULINT;
+}
+
+void FORTE_influxDB__Timestamp::executeEvent(const TEventID paEIID, CEventChainExecutionThread *const paECET) {
+  switch(paEIID) {
+    case scmEventREQID:
+        var_timestamp = CIEC_ULINT(getNanoSecondsRealtime());
```

Comment:
at least for the win32 platform the value returned by `getNanoSecondsRealtime()` is (at least for now) not related to the UNIX epoch

https://learn.microsoft.com/en-us/windows/win32/sysinfo/acquiring-high-resolution-time-stamps#general-faq-about-qpc-and-tsc